src/mappings/entities/mapClaim.ts

```diff
@@ -0,0 +1,148 @@
+  Jurisdiction = "Jurisdiction",
+  Exempted = "Exempted",
+  Blocked = "Blocked",
+  InvestorUniqueness = "InvestorUniqueness",
+  NoData = "NoData",
+  InvestorUniquenessV2 = "InvestorUniquenessV2",
+}
+
+type ClaimParams = {
+  claimExpiry: bigint;
+  claimIssuer: string;
+  claimScope: string;
+  claimType: string;
+  issuanceDate: bigint;
+  lastUpdateDate: bigint;
+  cddId: string;
+};
+
+/**
+ * Subscribes to the Claim events
+ */
+export async function mapClaim(
+  blockId: number,
+  eventId: EventIdEnum,
+  moduleId: ModuleIdEnum,
+  params: Codec[],
+  event: SubstrateEvent,
+  claimData: ClaimParams
+): Promise<void> {
+  if (moduleId === ModuleIdEnum.Identity && isClaimEvent(eventId)) {
+    let jurisdiction;
+    if (claimData.claimType === ClaimTypeEnum.Jurisdiction) {
+      const col1: string = JSON.parse(claimData.claimScope).col1;
+      jurisdiction = col1.substring(0, 2);
+    }
+
+    const targetDid = getTextValue(params[0]);
+
+    const scope = JSON.parse(claimData.claimScope);
+
+    const claim = {
+      targetDid,
+      issuer: claimData.claimIssuer,
+      issuanceDate: claimData.issuanceDate,
+      lastUpdateDate: claimData.lastUpdateDate,
+      expiry: claimData.claimExpiry,
+      type: claimData.claimType,
+      scope: JSON.parse(claimData.claimScope),
+      jurisdiction,
+      cddId: claimData.cddId,
+    };
+
+    await Claim.create({
+      id: `${blockId}/${event.idx}`,
+      ...claim,
+      filterExpiry: claimData.claimExpiry || "253402194600000",
+    }).save();
+
+    const identityWithClaims = await IdentityWithClaims.get(targetDid);
+    if (identityWithClaims) {
+      identityWithClaims.claims.push(claim);
+      await identityWithClaims.save();
+    } else {
+      await IdentityWithClaims.create({
+        id: targetDid,
+        did: targetDid,
+        claims: [claim],
+      }).save();
+    }
+
+    const issuerIdentityWithClaims = await IssuerIdentityWithClaims.get(
+      claimData.claimIssuer
+    );
+    if (issuerIdentityWithClaims) {
+      issuerIdentityWithClaims.claims.push(claim);
+      await issuerIdentityWithClaims.save();
+    } else {
+      await IssuerIdentityWithClaims.create({
+        id: claimData.claimIssuer,
+        did: claimData.claimIssuer,
+        claims: [claim],
+      }).save();
+    }
+
+    if (scope) {
+      handleScopes(
+        targetDid,
+        scope.type === ClaimScopeTypeEnum.Ticker ? scope.value : null,
+        scope
+      );
+    }
+  }
+
+  if (
+    moduleId === ModuleIdEnum.Identity &&
+    eventId === EventIdEnum.AssetDidRegistered
+  ) {
+    const targetDid = getTextValue(params[0]);
+    const ticker = serializeTicker(params[1]);
+    handleScopes(targetDid, ticker);
+  }
+}
+
+async function handleScopes(
+  targetDid: string,
+  ticker?: string,
+  scope?: { type: string; value: string }
+) {
+  const id = `${targetDid}/${scope?.value || ticker}`;
+  const scopeByDid = await ClaimScope.get(id);
+  if (!scopeByDid) {
```

Comment:
(nit) Since it seems like you have all the fields I think you can just call `create({...}).save()` and it will create an upsert that does nothing if its already there. One less call + slightly less code.